src/client/lazy-app/Compress/index.tsx

```diff
@@ -916,6 +963,110 @@ export default class Compress extends Component<Props, State> {
         throw err;
       }
     });
+
+    return await Promise.allSettled(jobs);
+  }
+
+  private downloadFromUrl(url: string, fileName: string = 'download.zip') {
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = fileName;
+    a.target = '_blank';
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+    URL.revokeObjectURL(url);
+  }
+  /**
+   * Download all using settings from one of the two sides by
+   * queuing each image in succession.
+   *
+   * This may cause flashing on the user's screen.
```

Comment:
Yeah, we should not go through multiple UI state changes just to do computational work. This may be finally the reason to refactor `updateImage` (which is called by `queueUpdateImage`) to separate the computational and caching parts from triggering state changes.

Reply:
Sounds like a good plan. Do you think this is enough work to justify doing it in a separate PR? Refactoring the computational part is going to affect existing functionality too, but I otherwise understand the need to for a polished UX before releasing.

Reply:
Yeah I understand. How do you feel about stacking PRs? We could declare this branch the new root for this feature, and fork off two branches — one for UX improvements on one for the computational refactor. How does that sound? (And it’s completely okay to say “that’s sounds shit”)

Reply:
lol I have no preference in that regard so if that's how you prefer to work then sure. I guess time will tell whether that's going to be confusing or not.